awx/ui_next/src/components/MultiSelect/MultiSelect.jsx

```diff
@@ -92,19 +101,33 @@ class MultiSelect extends Component {
 
   handleSelection(e, item) {
     const { chipItems } = this.state;
-    const { onAddNewItem } = this.props;
+    const { onAddNewItem, onChange } = this.props;
     e.preventDefault();
 
+    const items = chipItems.concat({ name: item.name, id: item.id });
     this.setState({
-      chipItems: chipItems.concat({ name: item.name, id: item.id }),
+      chipItems: items,
       isExpanded: false,
     });
     onAddNewItem(item);
+    onChange(items);
+  }
+
+  createNewItem(name) {
+    const { createNewItem } = this.props;
+    if (createNewItem) {
+      return createNewItem(name);
+    }
+    return {
+      id: Math.random(),
```

Comment:
Math.random() produces unstable keys. In the case of job tags and skip tags, since those must always be unique we could use those values as keys.